packages/utils/node-resolver-core/src/NodeResolver.js

```diff
@@ -798,18 +803,27 @@ export default class NodeResolver {
       return null;
     }
 
-    let pkgKeys = ['source', 'alias'];
-    if (env.isBrowser()) pkgKeys.push('browser');
+    if (pkg.source && !Array.isArray(pkg.source)) {
```

Comment:
Making this function use static references rather than dynamic property accesses seems to improve performance quite a bit